iot/api/CloudIotMqttExample/Program.cs

```diff
@@ -0,0 +1,645 @@
+            int mqttBridgePort, int jwtExpiresMin, int waitTime)
+        {
+            //setting default values if not passed from parameters
+            numMsgs = numMsgs <= 0 ? 20 : numMsgs;
+            messageType = (!string.IsNullOrEmpty(messageType)) ? messageType : "events";
+            mqttBridgeHostName = (!string.IsNullOrEmpty(mqttBridgeHostName)) ? mqttBridgeHostName : "mqtt.googleapis.com";
+            mqttBridgePort = (mqttBridgePort != 8883 || mqttBridgePort != 443) ? 8883 : mqttBridgePort;
+            jwtExpiresMin = jwtExpiresMin <= 0 ? 20 : jwtExpiresMin;
+
+            MqttClient client = GetClient(projectId, cloudRegion, registryId, deviceId, privateKeyFile, algorithm, caCert, mqttBridgeHostName, mqttBridgePort);
+
+            // Create our MQTT client. The client_id is a unique string that identifies
+            // this device.For Google Cloud IoT Core, it must be in the format below.
+            var clientId = $"projects/{projectId}/locations/{cloudRegion}/registries/{registryId}/devices/{deviceId}";
+
+            // With Google Cloud IoT Core, the username field is ignored, and the
+            // password field is used to transmit a JWT to authorize the device.
+            DateTime iat = DateTime.Now;
+            var pass = CreateJwtRsa(projectId, privateKeyFile);
+
+            double initialConnectIntervalMillis = 0.5;
+            double maxConnectIntervalMillis = 6;
+            double maxConnectRetryTimeElapsedMillis = 900;
+            double intervalMultiplier = 1.5;
+
+            double retryIntervalMs = initialConnectIntervalMillis;
+            double totalRetryTimeMs = 0;
+
+            while (!client.IsConnected && totalRetryTimeMs < maxConnectRetryTimeElapsedMillis)
+            {
+                try
+                {
+                    // Connect to the Google MQTT bridge.
+                    client.Connect(clientId, "unused", pass);
+                }
+                catch (MqttCommunicationException ex)
+                {
+                    if (ex.InnerException != null)
+                    {
+                        Console.WriteLine("An error occured {0}", ex.InnerException.Message);
+                    }
+                    else
+                    {
+                        Console.WriteLine("An error occured {0}", ex.Message);
+                    }
+
+                    Console.WriteLine("Retrying in " + retryIntervalMs + " seconds.");
+
+                    System.Threading.Thread.Sleep((int)retryIntervalMs);
+                    totalRetryTimeMs += retryIntervalMs;
+                    retryIntervalMs *= 1.5;
+                    if (retryIntervalMs > maxConnectIntervalMillis)
+                    {
+                        retryIntervalMs = maxConnectIntervalMillis;
+                    }
+                }
+                catch (MqttClientException ex)
+                {
+                    Console.WriteLine("Client Exception" + ex.InnerException.Message);
```

Comment:
A newline in the string would probably make the message more readable when it's printed to stdout.